appserver/ejb/ejb-http-remoting/client/src/main/java/fish/payara/ejb/http/client/RemoteEJBContextFactory.java

```diff
@@ -77,9 +77,35 @@
     public static final String FISH_PAYARA_EXECUTOR_SERVICE = "fish.payara.executorService";
     public static final String FISH_PAYARA_CONNECT_TIMEOUT = "fish.payara.connectTimeout";
 
+    /**
+     * The keys checked when creating a {@link Context} with {@link #getInitialContext(Hashtable)}. If these are not set
+     * in the environment we they are initialised to a {@link System#getProperty(String)}. The name of the property is
+     * the same except {@code fish.parara.}/{@code java.naming.} is replaced with {@code fish.payara.ejb.http.client}
+     * effectively looking for properties with names like {@code fish.payara.ejb.http.client.sslContext} or
+     * {@code fish.payara.ejb.http.client.provider.url}.
+     */
+    private String[] SYSTEM_PROPERTY_KEYS = { Context.PROVIDER_URL, Context.SECURITY_CREDENTIALS,
+            Context.SECURITY_PRINCIPAL, FISH_PAYARA_CONNECT_TIMEOUT, FISH_PAYARA_EXECUTOR_SERVICE,
+            FISH_PAYARA_HOSTNAME_VERIFIER, FISH_PAYARA_KEY_STORE, FISH_PAYARA_READ_TIMEOUT,
+            FISH_PAYARA_SCHEDULED_EXECUTOR_SERVICE, FISH_PAYARA_SSL_CONTEXT, FISH_PAYARA_TRUST_STORE,
+            FISH_PAYARA_WITH_CONFIG };
+
+    @SuppressWarnings("unchecked")
     @Override
     public Context getInitialContext(Hashtable<?, ?> environment) throws NamingException {
+        updateEnvironmentFromSystemProperties((Hashtable<String, Object>) environment);
         return new RemoteEJBContext(environment);
     }
-    
+
+    private void updateEnvironmentFromSystemProperties(Hashtable<String, Object> environment) {
+        for (String key : SYSTEM_PROPERTY_KEYS) {
+            if (!environment.containsKey(key)) {
+                String systemPropertyName = key.replaceFirst("fish\\.payara\\.|java\\.naming\\.", "fish.payara.ejb.http.client.");
```

Comment:
I am starting to think that what users expect is that system properties of the same name as the environment variables are checked. Can anyone confirm?